cabal-install/Distribution/Client/FileMonitor.hs

```diff
@@ -199,7 +199,7 @@ monitorFileHashedSearchPath notFoundAtPaths foundAtPath =
 -- files to be monitored (index by their path), and a list of
 -- globs, which monitor may files at once.
 data MonitorStateFileSet
-   = MonitorStateFileSet !(Map FilePath MonitorStateFile)
+   = MonitorStateFileSet ![MonitorStateFile]
```

Comment:
I suggest it, only because it makes it semantically clear what its morally supposed to represent. Encode invariants in the types! Without a type, needs a comment.